flink-tests/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterStopWithSavepointIT.java

```diff
@@ -357,4 +391,36 @@ public void finishTask() throws Exception {
 			finishLatch.trigger();
 		}
 	}
+
+	/**
+	 * A {@link StreamTask} that simply calls {@link CountDownLatch#countDown()} when
+	 * invoking {@link #triggerCheckpointAsync}.
+	 */
+	public static class CheckpointCountingTask extends NoOpStreamTask {
```

Comment:
By extracting I meant that `ExceptionOnCallbackStreamTask` could also use this `CheckpointCountingTask` class to avoid some (minor) code duplication. Could you re-use this code in `ExceptionOnCallbackStreamTask`? Make it extend `CheckpointCountingTask` and delegate methods/constructors to `super` calls?

Reply:
Ok, will update it. previously did not change to this, because I though in `CheckpointCountingTask` we do not need to distinguish whether this task is in subtask 0 (and can tolerate some code dup in test), but distinguish whether this task is subtask 0 seems has no side effect.